datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java

```diff
@@ -87,10 +101,15 @@ public FlowIdentifier triggerSdxRepairFlow(SdxCluster cluster, SdxRepairRequest
 
     public FlowIdentifier triggerDatalakeRuntimeUpgradeFlow(SdxCluster cluster, String imageId, SdxUpgradeReplaceVms replaceVms) {
         LOGGER.info("Trigger Datalake runtime upgrade for: {} with imageId: {} and replace vm param: {}", cluster, imageId, replaceVms);
-        String selector = DATALAKE_UPGRADE_EVENT.event();
         String userId = ThreadBasedUserCrnProvider.getUserCrn();
-        return notify(selector, new DatalakeUpgradeStartEvent(selector, cluster.getId(),
+        if (entitlementService.publicDatalakeBackupOnUpgradeEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
```

Comment:
as datalakedr service could be disabled I think it should be also checked here next to the entitlement

Reply:
datatalake-dr service will always be running.

Reply:
you are right. datatalake-dr service might not be up in the dev environment.